junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java

```diff
@@ -52,7 +52,7 @@
  * @see TestExecutionListener
  */
 @API(status = STABLE, since = "1.0")
-public final class TestPlan {
+public class TestPlan {
```

Comment:
@marcphilipp Have you considered making `UnmodifiableTest` it's own separate class that just delegates to a `TestPlan` rather than extending it, so that `TestPlan` can be kept final?

Reply:
How would that help? `UnmodifiableTestPlan` must be an "instance of" `TestPlan` in order for JUnit to return it from `org.junit.platform.launcher.Launcher.discover(LauncherDiscoveryRequest)`.

Reply:
Good question @sbrannen! I admit I didn't have this PR checked out on my computer when I wrote my suggestion, so I missed that `Launcher.discover(LauncherDiscoveryRequest)` needed to return a plain `TestPlan`. Apologies! :bowing_man: 

In that case, I don't see anyway of allowing `TestPlan` to be kept final, apart from giving `TestPlan` a constructor that accepts a parameter which determines if the `TestPlan` is modifiable or not, but it's not clear to me that that's any cleaner than the current solution.